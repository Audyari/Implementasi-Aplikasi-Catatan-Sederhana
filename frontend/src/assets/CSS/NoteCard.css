/* NoteCard.css */
.note-card {
    border: 1px solid #e0e0e0;
    border-radius: 8px;
    padding: 16px;
    margin-bottom: 16px;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  }
  
  .note-card__header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 12px;
  }
  
  .note-card__title {
    margin: 0;
    font-size: 1.25rem;
    color: #333;
  }
  
  .note-card__date {
    color: #666;
    font-size: 0.875rem;
  }
  
  .note-card__content {
    color: #444;
    margin-bottom: 16px;
    line-height: 1.5;
  }
  
  .note-card__delete {
    background-color: #ff4444;
    color: white;
    border: none;
    padding: 8px 16px;
    border-radius: 4px;
    cursor: pointer;
    font-size: 0.875rem;
    transition: background-color 0.2s;
  }
  
  .note-card__delete:hover {
    background-color: #cc0000;
  }