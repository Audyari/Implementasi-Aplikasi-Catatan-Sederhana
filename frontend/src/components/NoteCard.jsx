import React from 'react';
import PropTypes from 'prop-types';
import '../assets/CSS/NoteCard.css';

const NoteCard = ({ title, content, date, onDelete }) => {
  // Format tanggal menjadi format yang lebih mudah dibaca
  const formattedDate = new Date(date).toLocaleDateString('id-ID', {
    day: 'numeric',
    month: 'long',
    year: 'numeric',
    hour: '2-digit',
    minute: '2-digit'
  });

  return (
    <div className="note-card">
      <div className="note-card__header">
        <h3 className="note-card__title">{title}</h3>
        <small className="note-card__date">{formattedDate}</small>
      </div>
      <div className="note-card__content">
        {content.length > 100 ? `${content.substring(0, 100)}...` : content}
      </div>
      {onDelete && (
        <div className="note-card__actions">
         <button 
            className="note-card__delete" 
            onClick={onDelete}
            aria-label="Hapus catatan"
            data-testid="delete-button"  // Tambahkan ini
        >
            Hapus
        </button>
        </div>
      )}
    </div>
  );
};

NoteCard.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  date: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.instanceOf(Date)
  ]).isRequired,
  onDelete: PropTypes.func
};

export default NoteCard;