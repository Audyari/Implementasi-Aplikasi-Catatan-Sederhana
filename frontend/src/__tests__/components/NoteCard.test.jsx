import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import NoteCard from '../../components/NoteCard';

describe('NoteCard Component', () => {
  const mockOnDelete = jest.fn();
  const defaultProps = {
    id: 1,
    title: 'Test Title',
    content: 'This is a test content for the note card component.',
    date: '2025-06-14T10:00:00Z',
    onDelete: mockOnDelete,
  };

  beforeEach(() => {
    mockOnDelete.mockClear();
  });

  it('renders the note title', () => {
    render(<NoteCard {...defaultProps} />);
    expect(screen.getByText(defaultProps.title)).toBeInTheDocument();
  });

  it('formats and displays the date correctly', () => {
    render(<NoteCard {...defaultProps} />);
    // Check for the formatted date (adjust the format based on your locale)
    expect(screen.getByText(/14 Juni 2025/)).toBeInTheDocument();
  });

  it('displays truncated content when longer than 100 characters', () => {
    const longContent = 'a'.repeat(150);
    render(<NoteCard {...defaultProps} content={longContent} />);
    
    const displayedContent = screen.getByText(/^a{100}/);
    expect(displayedContent).toBeInTheDocument();
    expect(displayedContent.textContent.endsWith('...')).toBe(true);
  });

  it('displays full content when 100 characters or less', () => {
    const shortContent = 'a'.repeat(100);
    render(<NoteCard {...defaultProps} content={shortContent} />);
    
    const displayedContent = screen.getByText(shortContent);
    expect(displayedContent).toBeInTheDocument();
    expect(displayedContent.textContent.endsWith('...')).toBe(false);
  });

  it('calls onDelete with note id when delete button is clicked', () => {
    render(<NoteCard {...defaultProps} />);
    
    const deleteButton = screen.getByRole('button', { name: /hapus/i });
    fireEvent.click(deleteButton);
    
    expect(mockOnDelete).toHaveBeenCalledTimes(1);
  });

  it('renders correctly without onDelete prop', () => {
    const { title, content, date } = defaultProps;
    render(<NoteCard title={title} content={content} date={date} />);
    
    expect(screen.getByText(title)).toBeInTheDocument();
    expect(screen.queryByRole('button', { name: /hapus/i })).not.toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<NoteCard {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
