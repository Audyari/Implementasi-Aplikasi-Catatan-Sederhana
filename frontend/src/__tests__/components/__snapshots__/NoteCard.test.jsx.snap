// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteCard Component matches snapshot 1`] = `
<div>
  <div
    class="note-card"
  >
    <div
      class="note-card__header"
    >
      <h3
        class="note-card__title"
      >
        Test Title
      </h3>
      <small
        class="note-card__date"
      >
        14 Juni 2025 pukul 17.00
      </small>
    </div>
    <div
      class="note-card__content"
    >
      This is a test content for the note card component.
    </div>
    <div
      class="note-card__actions"
    >
      <button
        aria-label="Hapus catatan"
        class="note-card__delete"
      >
        Hapus
      </button>
    </div>
  </div>
</div>
`;
